From 051d1a079ca4e64ed5468aa276b6d194c0b6181f Mon Sep 17 00:00:00 2001
From: Dipen Dudhat <dipen.dudhat@freescale.com>
Date: Thu, 29 Oct 2009 22:26:51 +0530
Subject: [PATCH ] Enable env support for SD and eSPI boot on P1 and P2 RDB platforms

Enable environment support for eSDHC/eSPI boot on P1 and P2 RDB platforms.

Signed-off-by: Dipen Dudhat <dipen.dudhat@freescale.com>
---
 include/configs/P1_P2_RDB.h |   16 ++++++++++++----
 1 files changed, 12 insertions(+), 4 deletions(-)

diff --git a/include/configs/P1_P2_RDB.h b/include/configs/P1_P2_RDB.h
index c93fdb5..25882e3 100644
--- a/include/configs/P1_P2_RDB.h
+++ b/include/configs/P1_P2_RDB.h
@@ -491,10 +491,18 @@ extern unsigned long get_board_sys_clk(unsigned long dummy);
 	#define CONFIG_ENV_IS_IN_NAND	1
 	#define CONFIG_ENV_SIZE		CONFIG_SYS_NAND_BLOCK_SIZE
 	#define CONFIG_ENV_OFFSET	((512 * 1024) + CONFIG_SYS_NAND_BLOCK_SIZE)
-#elif defined(CONFIG_RAMBOOT_SDCARD) || defined(CONFIG_RAMBOOT_SPIFLASH)
-	#define CONFIG_ENV_IS_NOWHERE	1	/* Store ENV in memory only */
-	#define CONFIG_ENV_ADDR		(CONFIG_SYS_MONITOR_BASE - 0x1000)
-	#define CONFIG_ENV_SIZE		0x2000
+#elif defined(CONFIG_RAMBOOT_SPIFLASH)
+	#define CONFIG_ENV_IS_IN_SPI_FLASH	1
+	#define CONFIG_ENV_SPI_BUS		0
+	#define CONFIG_ENV_SPI_CS		0
+	#define CONFIG_ENV_SPI_MAX_HZ		10000000
+	#define CONFIG_ENV_SPI_MODE		0
+	#define CONFIG_ENV_SIZE			0x2000		/* 8KB */
+	#define CONFIG_ENV_OFFSET		0x100000	/* 1MB */
+	#define CONFIG_ENV_SECT_SIZE		0x10000
+#elif defined(CONFIG_RAMBOOT_SDCARD)
+	#define CONFIG_ENV_IS_IN_SDCARD		1
+	#define CONFIG_ENV_SIZE			0x2000
 #endif
 #else
 	#define CONFIG_ENV_IS_IN_FLASH	1
-- 
1.5.6.3

